### ISSUE TITLE
Regression is updates between 3.4.9 and 3.4.10

### ISSUE THREAD

**Issue by CrackerJackMack** [Exa-Networks/exabgp#276]
Been running exabgp for a few months and noticed a regression when I updated to 3.4.10.  Looks like it might be an encoding issue related to asn4 ?  Meant to file this a while back but  just getting around to it :calendar: has been swamped.  Downgrading back to 3.4.9 fixed my issue for the time being and holding off on upgrading.  This is taken from my production systems so I don't have any packet dumps available for you but will try to provide enough information to recreate in the lab.

Partially sanitized.

Router
--------
``` 
System Mode: XMR
Chassis: NetIron 16-slot
```

Router config (SOP)
-----------------
```
route-map deny-all deny 10
!
!
ip prefix-list something-anycast-ipv4-range seq 5 permit 10.255.0.49/32
ip as-path access-list something-anycast-asn seq 5 permit _64902_
!
route-map  something-anycast-ipv4-in permit 10
    match ip address prefix-list something-anycast-ipv4-range
    match as-path something-anycast-asn
    set metric 0
!
router bgp
    neighbor SOMETHING-ANYCAST-IPV4 peer-group
    neighbor SOMETHING-ANYCAST-IPV4 remote-as 64902
    neighbor SOMETHING-ANYCAST-IPV4 remove-private-as
    neighbor SOMETHING-ANYCAST-IPV4 description "INTERNAL something anycast"
    neighbor SOMETHING-ANYCAST-IPV4 password $BGP_MD5PASSWORD
    neighbor $EBGP_IPV4_NEIGHBOR peer-group SOMETHING-ANYCAST-IPV4
    address-family ipv4 unicast
        neighbor SOMETHING-ANYCAST-IPV4 route-map in  something-anycast-ipv4-in
        neighbor SOMETHING-ANYCAST-IPV4 route-map out deny-all
        neighbor SOMETHING-ANYCAST-IPV4 activate
    exit-address-family
exit
!
```

Exabgp config (for both 3.4.9 and 3.4.10)
-----------------------------------------------------

```bind
group something {
    description "something";
    router-id 10.252.79.4;

    local-address 10.252.79.4;
    local-as 64902;
    peer-as 65000;
    md5 "xxx45xxxx";

    family {
        ipv4 unicast;
    }

    neighbor 10.252.79.2 {
    }

    neighbor 10.252.79.3 {
    }

    process watch-something {
        encoder text;
        run "/usr/local/bin/something_health.sh";
    }
}
```

something_health.sh
---------------------------
```bash
#!/bin/bash
anycast=10.255.0.49
# stripped all healthcheck code and 
echo "announce route $anycast next-hop self"
```

Relevant output for 3.4.10
-----------------------
```
Fri, 08 May 2015 18:35:09 | INFO     | 21347  | network       | Connected to peer neighbor 10.252.79.3 local-ip 10.252.79.4 local-as 64902 peer-as 65000 router-id 10.252.79.4 family-allowed in-open (out)
Fri, 08 May 2015 18:35:09 | INFO     | 21347  | network       | Peer     10.252.79.3 ASN 65000 out loop, peer reset, message [notification received (3,11)] error[UPDATE message error / Malformed AS_PATH]
Fri, 08 May 2015 18:35:13 | INFO     | 21347  | network       | Connected to peer neighbor 10.252.79.2 local-ip 10.252.79.4 local-as 64902 peer-as 65000 router-id 10.252.79.4 family-allowed in-open (out)
Fri, 08 May 2015 18:35:13 | INFO     | 21347  | network       | Peer     10.252.79.2 ASN 65000   out loop, peer reset, message [notification received (3,11)] error[UPDATE message error / Malformed AS_PATH]
Fri, 08 May 2015 18:36:14 | INFO     | 21347  | network       | Peer     10.252.79.3 ASN 65000   connection to peer failed
Fri, 08 May 2015 18:36:18 | INFO     | 21347  | network       | Peer     10.252.79.2 ASN 65000   connection to peer failed
Fri, 08 May 2015 18:37:19 | INFO     | 21347  | network       | Peer     10.252.79.3 ASN 65000   connection to peer failed
Fri, 08 May 2015 18:37:23 | INFO     | 21347  | network       | Peer     10.252.79.2 ASN 65000   connection to peer failed
```

**Comment by thomas-mangin** [comment]
Thank you for the bug report, this should be fixed in the 3.4 branch on github. I will release a 3.4.11 version soon.

**Comment by CrackerJackMack** [comment]
Ah, found it.  I couldn't find #250 prior.  Thank you for the fix.

**Comment by thomas-mangin** [comment]
I have just released 3.4.11.

**Comment by CrackerJackMack** [comment]
3.4.11 fixed it.  Thank you so much.
